src/sbvr-api/permissions.ts

```diff
@@ -298,30 +298,47 @@ const collapsePermissionFilters = <T>(
 	return v;
 };
 
-const namespaceRelationships = (
+function namespaceRelationships(
+	relationships: RelationshipInternalNode,
+	alias: string,
+): undefined | RelationshipInternalNode;
+function namespaceRelationships(
+	relationships: Relationship,
+	alias: string,
+): undefined | Relationship;
+function namespaceRelationships(
```

Comment:
If I understand correctly, `Relationship` is simply a union type over Internal and LeafNodes. Wouldn't
```suggestion
function namespaceRelationships<T extends Relationship>(
	relationships: T,
	alias: string,
): undefined | T {
```

Be the same and more concise?